Tests/AppTests/PackageController+routesTests.swift

```diff
@@ -17,107 +17,87 @@
 import Vapor
 import XCTest
 
-class PackageControllerTests: AppTestCase {
+class PackageController_routesTests: AppTestCase {
 
     func test_show() throws {
         // setup
         let pkg = try savePackage(on: app.db, "1")
         try Repository(package: pkg, name: "package", owner: "owner")
             .save(on: app.db).wait()
+        try Version(package: pkg, latest: .defaultBranch).save(on: app.db).wait()
```

Comment:
This is a new requirement: a package _must_ have a default branch version. This is true for all packages, i.e.

```sql
select p.url
from packages p
left join versions v on v.package_id = p.id
where v.id is null
```

returns no results on dev and prod.

I'd say it's ok to impose this. A package without a default version would be a package without any commits (or at least without any commits we can parse for some reason). 404ing on that would be ok even if it's a problem on our end, because we'd want to/need to fix that.

Alternatively, it should be possible to drop this requirement again but it'd require some digging into the mechanics of the underlying query and re-introduction of optionals in quite a few places.

Reply:
I agree with the requirement completely, and can't see a situation where a package would not have a default branch. Is that even possible with git?

It's not clear to mr wheat would happen if we *did* somehow get a package in there with a default branch. What would the effect on that package in the index?

Reply:
No, that wouldn’t be possible with git. But it could theoretically happen if for instance our analyser crashed while getting the versions or if for some reason.

If that did happen, the package would not show up in any lists and would 404 if you navigated to its url.